File: src/frontend/src/dtos/PlaylistDto.ts
import {TrackDto} from "@/dtos/TrackDto";

export interface PlaylistDto {
    title: string;
    author?: string;
    url?: string;
    artworkUrl?: string | null;
    trackCount?: number;
    selectedTrack?: TrackDto;
    encodedTracks?: string[];
}

File: src/DiscordMusicBot/Dtos/PlaylistDto.cs
namespace DiscordMusicBot.Dtos;

public record struct PlaylistDto
{
    public string Title { get; init; }
    public string? Author { get; init; }
    public string? Url { get; init; }
    public Uri? ArtworkUrl { get; init; }
    public int? TrackCount { get; set; }
    public TrackDto? SelectedTrack { get; set; }
    public string[]? EncodedTracks { get; set; }
}